subscriber.go

```diff
@@ -486,26 +492,38 @@ func (s *Subscriber) Sync(ctx context.Context, peerID peer.ID, nextCid cid.Cid,
 		return cid.Undef, err
 	}
 
-	hnd.syncMutex.Lock()
-	defer hnd.syncMutex.Unlock()
+	if updateLatest {
+		// Grab the latestSyncMu lock so that an async handler doesn't update the
+		// latestSync between when we call hnd.handle and when we actually updateLatest.
+		hnd.latestSyncMu.Lock()
+		defer hnd.latestSyncMu.Unlock()
+	}
 
-	err = hnd.handle(ctx, nextCid, sel, wrapSel, updateLatest, syncer, cfg.scopedBlockHook)
+	syncedCids, err := hnd.handle(ctx, nextCid, sel, wrapSel, syncer, cfg.scopedBlockHook)
 	if err != nil {
 		return cid.Undef, fmt.Errorf("sync handler failed: %w", err)
 	}
 
+	if updateLatest {
+		hnd.latestSync = cidlink.Link{Cid: nextCid}
+		hnd.subscriber.inEvents <- SyncFinished{Cid: nextCid, PeerID: hnd.peerID, SyncedCids: syncedCids}
+		log.Infow("Updating latest sync")
```

Comment:
make this log more useful: include peer, cid

Reply:
That is already done by `log.With` above.